test/unit/TestServer.ts

```diff
@@ -0,0 +1,45 @@
+import getPort = require('get-port');
+import { createServer, IncomingMessage, Server, ServerResponse } from 'http';
+import { URL } from 'url';
+
+export type RequestHandler = (
+  req: IncomingMessage,
+  res: ServerResponse
+) => void;
+
+export class RealTestServer {
```

Comment:
I didn’t look very hard for another package that implements something like this. I found mocking libraries for http but those always felt messy to me, and so I chose to have a real local server that could be hit for testing.